### PR TITLE
[Palo Alto] Error 401 correctly

### PR DESCRIPTION
**Pull request by nmccullagh-r7**
## Proposed Changes
https://rapid7.atlassian.net/browse/SOAR-18176

### CODE REVIEW

File: plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/tasks/monitor_alerts/task.py [hunk 1]
```diff
@@ -61,6 +61,15 @@ def run(self, params={}, state={}, custom_config: dict = {}):  # pylint: disable
 
             return response, state, has_more_pages, 200, None
 
+        except APIException as error:
+            return (
+                [],
+                {},
```
Comment (ekelly-r7) [inline plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/tasks/monitor_alerts/task.py:67]: Do we want to return an empty dict for state here? or should just return `existing_state` ?

File: plugins/palo_alto_cortex_xdr/unit_test/test_get_file_quarantine_status.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import logging
+
+
+class TestGetFileQuarantineStatus(TestCase):
+    def test_get_file_quarantine_status(self):
+        """
+        DO NOT USE PRODUCTION/SENSITIVE DATA FOR UNIT TESTS
+
+        TODO: Implement test cases here
+        """
```
Comment (ekelly-r7) [inline plugins/palo_alto_cortex_xdr/unit_test/test_get_file_quarantine_status.py:19]: prob worth removing this test file as it's just a stub?

File: plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/tasks/monitor_alerts/task.py (hunk 1 above)
Comment (nmccullagh-r7) [reply plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/tasks/monitor_alerts/task.py:67]: existing state isn't used for anything in an error case in pif 🤷

Comment (ekelly-r7) [reply plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/tasks/monitor_alerts/task.py:67]: yeah fair enough, just a np more than anything as that's the (albeit redundant) pattern that we usually follow

**Review by ekelly-r7 — APPROVED** [review]

File: plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/util/api.py [hunk 3]
```diff
@@ -473,12 +474,21 @@ def _handle_401(self, response: Response, url: str, post_body: dict):
...
-        if response.status_code == 401:
-            self.logger.info(f"Received HTTP {response.status_code}, re-authenticating to Palo Alto Cortex XDR")
+        if response.status_code in [401, 403]:
+            self.logger.info(f"Received HTTP {response.status_code}, attempting to refresh token")
 
             new_headers = self._advanced_authentication(api_key=self.api_key, api_key_id=self.api_key_id)
             new_response = self.build_request(url=url, headers=new_headers, post_body=post_body)
 
+            if new_response.status_code in [401, 403]:
+                self.logger.info("Token is still invalid, raising exception")
```
Comment (ekelly-r7) [inline plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/util/api.py:484]: np: I think should be `self.logger.error`

**Pull request merged**